File: JavaScript/heroi.js
class heroi {
    constructor(nome, idade, tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;

        switch (this.tipo) {
            case "mago":
                this.tipoAtaque = "magia";
                break;
            case "guerreiro":
                this.tipoAtaque = "espada";
                break;
            case "monge":
                this.tipoAtaque = "artes marciais";
                break;
            case "ninja":
                this.tipoAtaque = "shuriken";
                break;
        }
    }

    atacar() {
       return console.log(`O ${this.tipo} ${this.nome} atacou usando ${this.tipoAtaque}`);
    }
}

const juninho = new heroi("Juninho", 25, "mago");
const alberto = new heroi("Alberto", 34, "guerreiro");

juninho.atacar();
alberto.atacar();

File: Java/src/Heroi.java
public class Heroi {

    private String nome;
    private int idade;
    private TipoHeroi tipo;
    private String tipoAtaque;

    public Heroi(String nome, int idade, TipoHeroi tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;

        switch (this.tipo) {
            case MAGO: this.tipoAtaque = "magia";
            break;

            case GUERREIRO: this.tipoAtaque = "espada";
            break;

            case MONGE: this.tipoAtaque = "artes marciais";
            break;

            case NINJA: this.tipoAtaque = "shuriken";
            break;
        }
    }

    public void atacar() {
        System.out.printf("\nO %s %s atacou o alvo utilizando sua %s", tipo, nome, tipoAtaque);
    }
}